### ISSUE TITLE
Asterisk patches after v20

### ISSUE THREAD

**Issue by realJoshByrnes** [usecallmanagernz/patches#9]
Just wondering what the future holds for these patches after Asterisk 20?

It's clear from the codebase of Asterisk 21 that chan_sip is removed.

I guess the options would be to:

* Re-introduce chan_sip
* Move to chan_pjsip
* End the project

Have you put any thought into this?

**Comment by gareth-palmer** [comment]
I had look at porting the patch to chan_pjsip, however because the features there are split into separate modules it wouldn't work as for example; the phones expect subscriptions to be working during register.

The easiest option would be to port chan_sip to Asterisk 21 but remove any parts that the phones don't need.

**Comment by realJoshByrnes** [comment]
To strip back chan_sip it would probably be better to rename the transport to something like chan_ucm (or similar)

I've only been using chan_sip for Cisco devices for quite some time now. I'm guessing that would be the same for most people with Cisco devices.

**Comment by lachesis** [comment]
Are you or anyone else stepping up to maintain chan_sip in Asterisk 22+? Can you expand more on why it would not be possible to port to pjsip? Is it a matter of being a lot of work and no ability/time/motivation to do it, or an actual limitation in the pjsip architecture that makes it impossible to support Cisco's phones?

**Comment by gareth-palmer** [comment]
chan_pjsip is implements SIP as a set of separate modules that may or may not be loaded. It is possible to support registering endpoints but not subscriptions or mailboxes etc, also some of the request/response handling is done via generic callbacks that don't support endpoint specific behaviour.

Cisco phones need almost all of SIP to be available and in the cases where Cisco-specific extensions are required those callbacks will need to be modified to include an extra parameter.

That would take additional extra work over what it took for chan_sip and it would never be accepted upstream, at that point it would be better to create a new Cisco specific SIP channel driver that also uses libpjsip.

**Comment by tmittelstaedt** [comment]
> Are you or anyone else stepping up to maintain chan_sip in Asterisk 22+

Yes, a community Asterisk developer is maintaining chan_sip.  And it DOES still apply to Asterisk 22. (which just went to RC1)

The repository is located here:

https://github.com/InterLinked1/chan_sip

Note that the To-Do List on his repo's readme states:

"The usecallmanager patches are not currently incorporated here because they present a huge merge conflict."

This is not, in fact, true.

**Comment by gareth-palmer** [comment]
There is now a patch for Asterisk 22. It uses a version of chan_sip that has been modernised and updated.
